semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+        a EclSubsidence object will be used to manage the rest of the
+        overburden timeshift calculations.
+        """
+        case = os.path.splitext(eclbase)[0]
+        self._init_file = EclFile("%s.INIT" % case)
+        self._rst_file = EclFile("%s.UNRST" % case)
+        self._grid = EclGrid("%s.EGRID" % case, apply_mapaxes=mapaxes)
+
+        self.subsidence = EclSubsidence(self._grid, self._init_file)
+
+        self._seabed = seabed
+        self._youngs_modulus = youngs * 1e9
+        self._poisson_ratio = poisson
+        self._r_factor = rfactor
+        self._convention = convention
+        self._verbose = verbose
+
+        self._surface = res_surface = OTSResSurface(grid=self._grid, above=above)
+        if velocity_model is not None:
+            self._surface = OTSVelSurface(res_surface=res_surface, vcube=velocity_model)
+
+        self._restart_views = {}
+
+    def get_horizon(self):
+        return self._create_surface()
+
+    def _create_surface(self, z=None):
+        """
+        Generate irap surface
+
+        :param z: replace z values of surface
+        """
+        nx = self._surface.nx
+        ny = self._surface.ny
+        x = self._surface.x
+        y = self._surface.y
+        if z is None:
+            z = self._surface.z
+
+        xstart = np.min(x)
+        ystart = np.min(y)
+
+        if nx < 2 or ny < 2:
+            raise RuntimeError("Cannot create IRAP surface if nx or ny is <2")
+
+        xinc = (np.max(x) - xstart) / (nx - 1)
+        yinc = (np.max(y) - ystart) / (ny - 1)
+
+        surf = Surface(
+            nx=nx, ny=ny, xinc=xinc, yinc=yinc, xstart=xstart, ystart=ystart, angle=0
+        )
+
+        irap_x = np.empty(nx * ny)
+        irap_y = np.array(irap_x)
+
+        for i, s in enumerate(surf):
+            irap_x[i], irap_y[i] = surf.getXY(i)
+
+        # Interpolate vel grid to irap grid, should be the same apart from ordering
+        z = np.nan_to_num(z)
+        ip = CloughTocher2DInterpolator((x, y), z, fill_value=0)
+        irap_z = ip(irap_x, irap_y)
+
+        for i, _ in enumerate(surf):
+            surf[i] = irap_z[i]
+
+        return surf
+
+    def add_survey(self, name, date):
+        """The add_survey() method will register a survey at a specific date.
+
+        The name argument should be a unique string, this will later
+        be used when evaluating the elastic strain in the
+        overburden. The date should be python datetime.date()
+        instance, this date should be present as a report step in the
+        restart file - otherwise an exception will be raised.
+        """
+        restart_view = self._rst_file.restartView(sim_time=date)
+        self.subsidence.add_survey_PRESSURE(name, restart_view)
+
+        self._restart_views[name] = restart_view
+
+        return restart_view
+
+    @staticmethod
+    def _divide_negative_shift(ts):
+        for i, _ in enumerate(ts):
+            if ts[i] < 0:
+                ts[i] /= 5.0
```

Comment:
What does this do? What does the number 5 represent? (would've made more sense to have such a parameter an argument)

Reply:
Pretty it does what it says, but why 5 is a good question :) Can do it as parameter defaulting to 5

Reply:
I mean, this is such a special thing to do.. I think we at least should state it in the documentation. It isn't nullified, it's just divided by the number 5.

Reply:
The question's how important is the division factor and whether it should be exposed as a parameter.

Reply:
Wrote to an author of this some days back, but still no answer. Shall we just have it as is then?

Reply:
Appending the answer from Thomas:

Observations (even though they are few) indicate that time shifts linked with compaction are smaller than when linked with stretch. This is also reported by e.g. Hatchel and Bourne (2005), which obtained that `R_stretch / R_comp` is around 5 (where R_stretch denotes R-factor for stretch and R_comp denotes the R-factor for compaction). Hatchell and Bourne obtained R_stretch=5 for Valhall field and we (Røste and Ke, 2017) obtained R_stretch=20 for sandstone fields, but we both obtained that `R_stretch / R_comp ≈ 5.` This is why we have chosen `R_comp = R_stretch / 5.`
 
The physical reason for this is not fully understood, but we believe rocks undergoing stretch changes differently than rocks undergoing compaction.

Reply:
What Thomas is saying is certainly true, but ideally the number 5 should be a parameter that could be input (through the ots_parameters file). For instance on Kvitebjørn we definitely see the "compaction halo" surrounding the reservoir, and by tuning the R-factor for streching, we could improve the match with 4D data. That could be important of fields with smaller compartments, like Valemon.

Reply:
https://github.com/equinor/semeio/issues/190